### PR TITLE
Poo

### PR DESCRIPTION
**Pull request by gastonmatta16**

### CODE REVIEW

**Comment by gastonmatta16** [comment]
I think its safe, on the last push i just add excersice 1 and ignored autocreated files

**Comment by gastonmatta16** [comment]
I have a question, if i have a pull request open and i push, its like everything that i pushed, automatically adds to the pull?

File: topics/src/POO/Ex2/ConnectionFactory.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package POO.Ex2;
+
+public interface ConnectionFactory {
```
Comment (sebastianscatularo) [inline topics/src/POO/Ex2/ConnectionFactory.java]: you don't need a interface for this concept

File: topics/src/POO/Ex2/ConnectionDB.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package POO.Ex2;
+import java.sql.*;
+
+public interface ConnectionDB {
```
Comment (sebastianscatularo) [inline topics/src/POO/Ex2/ConnectionDB.java]: this could be renamed to Connection

File: topics/src/POO/Ex2/PostgreSQLConnectionFactory.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package POO.Ex2;
+
+public class PostgreSQLConnectionFactory implements ConnectionFactory {
```
Comment (sebastianscatularo) [inline topics/src/POO/Ex2/PostgreSQLConnectionFactory.java]: As Factory Pattern definition, you should be able to create an object instance of a family using the Factory, for example

```suggestion
public final class ConnectionFactory {
    private ConnectionFactory() { }
    public static Connection getConnection(String type) {
        switch(type) {
            "java": return Java.getInstance();
            "mysql": return Mysql.getInstance();
            ...
            default: return NoOp.getInstance();
        }
    }
}
```

File: topics/src/POO/Ex2/ConnectionDB.java (hunk 2 above)
Comment (gastonmatta16) [reply topics/src/POO/Ex2/ConnectionDB.java]: Done.

File: topics/src/POO/Ex2/PostgreSQLConnectionFactory.java (hunk 3 above)
Comment (gastonmatta16) [reply topics/src/POO/Ex2/PostgreSQLConnectionFactory.java]: Could we having a misunderstanding? this is the second point of the practice :
"Create an abstract factory example for the different type of SQL connections." 
this code that you showed me, isn't a Factory pattern? not AbstractFactory or am i wrong?

Comment (gastonmatta16) [reply topics/src/POO/Ex2/PostgreSQLConnectionFactory.java]: this are the links from where i reading
https://www.oodesign.com/factory-method-pattern.html
https://www.oodesign.com/abstract-factory-pattern.html

Comment (sebastianscatularo) [reply topics/src/POO/Ex2/PostgreSQLConnectionFactory.java]: wee you could also refer to https://www.oodesign.com/factory-pattern.html and there is a producId, but that product id could be anything that make sense in your context

**Pull request merged**